index.js

```diff
@@ -45,8 +47,26 @@ function utm(query){
   return results;
 }
 
+var allowedKeys = ['name', 'term', 'source', 'medium', 'content'];
+
+/**
+ * Get strict utm params - from the given `querystring`
+ *
+ * @param {String} query
+ * @return {Object}
+ * @api private
+ */
+
+function strict(query) {
+  return foldl(function(acc, val, key) {
+    if (includes(key, allowedKeys)) acc[key] = val;
```

Comment:
This is a worst-case `n^2` operation, use a hash table to make this more efficient:

```js
var allowedKeys = {
  name: true,
  term: true,
  source: true,
  medium: true,
  content: true
};

//...
    if (allowedKeys.hasOwnProperty(key)) acc[key] = val;
```